vllm/worker/model_runner.py

```diff
@@ -1379,6 +1385,16 @@ def execute_model(
             logits=logits,
             sampling_metadata=model_input.sampling_metadata,
         )
+        if (self.observability_config.collect_model_forward_time
+                and output is not None):
+            model_forward_end.synchronize()
+            model_forward_time = model_forward_start.elapsed_time(
+                model_forward_end)
+            # If there are multiple workers, we are still tracking the latency
+            # from the start time of the driver worker to the end time of the
+            # driver worker. The model forward time will then end up covering
+            # the communication time as well.
+            output.model_forward_time = model_forward_time
```

Comment:
does it work with PP case?

Reply:
Should I test the PP case ? Can you give me a command line or model that I can do this with ? I have tested it with tensor parallelism on the llama models. I don't yet fully understand how all the code is laid out to be confident where the request processing joins in on the PP paths.